Cds 67121 cds 71172 post prod rollback (#2177)
harness/developer-hub: docs/continuous-delivery/x-platform-cd-features/advanced/rollback-deployments.md

---
title: Rollback deployments
description: Rollback your most recent successful deployment.
sidebar_position: 3
---

:::info

Currently, this feature is behind the feature flag, `POST_PROD_ROLLBACK`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.

:::

Rollback deployment initiates a rollback of your most-recent successful deployment. This allows rapid, predictable recovery from a deployment that succeeded on technical criteria, but that you want to undo for other reasons.

## Important notes

* Rollback deployment is currently supported for the following deployment types only: 
  - Kubernetes
  - Tanzu Application Services (TAS)

    Harness anticipates expanding this feature to other deployment types in the future.
* You can rollback successful pipelines only. The rollback option is not available for failed pipelines.
* Only the rollback steps that are part of the stage can be rolled back.

> * You cannot rollback the same pipeline multiple times. It can be rolled back if the pipeline is executed again.

Author: ```suggestion
* You cannot rollback the same pipeline multiple times. You can perform rollback if the pipeline is executed again.
```

* If the pipeline configuration has changed between executions, the previous execution YAML is used as reference to rollback.

> * If the rollback steps of a stage have the same expressions present in the steps of a successful deployment, the steps of the deployed pipeline must be resolved first. For example, you should be able to reuse a step in the rollback section as the output of a step in the steps section.

Author: @sahilhindwani-harness I need some clarity and confirmation on the technical accuracy of this point.

Reviewer: Looks good to me

Reviewer: We can refer an expression of normal step executed during pipeline execution in the rollback section as well and it should be automatically resolved with that value from original execution incase of post prod rollback. 

Maybe we can write the above as our last point.

> * If the rollback steps of a stage have the same expressions present in the steps of a successful deployment, the steps of the deployed pipeline must be resolved first. For example, you should be able to reuse a step in the rollback section as the output of a step in the steps section.

Author: ```suggestion
* You can refer expressions used in a step executed during the pipeline execution in the deployment rollback step as well. The expressions are automatically resolved with values from the original execution in the deployment rollback step.
```
